src/Adapter/MSTest.CoreAdapter/Execution/TestClassInfo.cs

```diff
@@ -230,8 +243,38 @@ public void RunClassInitialize(TestContext testContext)
                 throw new NullReferenceException(Resource.TestContextIsNull);
             }
 
-            // If class initialization is not done, then do it.
-            if (!this.IsClassInitializeExecuted)
+            MethodInfo initializeMethod = null;
+            string failedClassInitializeMethodName = string.Empty;
+
+            try
+            {
+                // If class initialization is not done, we can proceed to perform
+                // base class initialization first
+                if (!this.IsClassInitializeExecuted)
+                {
```

Comment:
nit: might be cleaner to just return from this method if `isClassInitializeExecuted`

Reply:
true, i was following same pattern as few lines below but that way seems little bit cleaner. Changes coming up in next push.